src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,85 @@
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string;
+
     @Override
     public String getString() {
-        return null;
+        return this.string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        //If string is null or blank, return 0.
+        if(this.string == null || this.string.isBlank()) return 0;
+        /*
+        Otherwise, get the number of words in string, which are
+        separated by one or more whitespace characters.
+         */
+        return string.split("[\\s\\r\\t\\n]+").length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        //check null string
+        if(this.string == null) return null;
+        // check n <= 0
+        if(n <= 0) throw new IllegalArgumentException("N must be greater than 0");
+        // check n > string string
+        if(n > this.string.length()) throw new IndexOutOfBoundsException("Index out of bounds");
+        //edge cases
+        if(n == 1 && !maintainSpacing) return "";
+        if(n == 1 && maintainSpacing) return this.string.replaceAll(".", " ");
```

Comment:
you could make this as an inbulit feature   of the for loop to save space  and handle edge case